### ISSUE TITLE
[BUG] Slow Package Resolve

### ISSUE THREAD

**Issue by JotaPe** [vrchat-community/creator-companion#458]
### Describe the bug

Creator Companion is incredibly slow as more registries you add, and it doesn't cache registry resolves, so instead of being snappy to manage your dependencies, every time you open your manage project page, you have to resolve all the registries before doing anything. Which would be fine if it wasn't doing everytime I go back to the manage project page, it doesn't cache the registry resolve and the package resolve too, the last on I understand, because a user can remove a package on Unity and break the state between the applications, but taking alot of time to see if I need to update a package? I think this should be fixed by caching most of the data that's hard to be broken by user behaviour, like the package versions, which packages the registry have, this should be atleast cached between Creator Companion sessions and if you hit the refresh button, it should force that cache to be reloaded.

And no, filtering the registries for each project doesn't fix this problem.

I don't think this is being caused by a slow drive too? Could be, it's on a NVME Sata SSD, which I don't think is the bottleneck here.

### To Reproduce

1. Open Creator Companion
2. Have lots of registries
3. Open a manage project from a project, wait to resolve
4. Open another manage project from another project

### Expected behavior

Should not take a lot of time to resolve the packages, should be instant for checking if I need to update something.

### Screenshots

Images are better than  words, multiple images in sequence even better!
https://github.com/vrchat-community/creator-companion/assets/17516509/1e076fc1-a51d-4ac4-bb2c-ecbaaba19a87

### Related Product

VCC (VRChat Creator Companion)

### Product Version

2.2.5

### OS

Windows 10

### Additional context

I totally forget that this project is Open Source, I will take a look to profile this if I get time for it.

**Comment by flaminglynx** [comment]
Hello, I think I am having the same issue but I am not as well versed in this stuff. I just had my motherboard replaced and when I got my PC back it was acting like a new PC but still had all my files and settings on it. Windows needed to be reactivated and whatnot. I'm having the slowness with creator companion and with unity as well which I didn't have before. Could this be something to do with the registries? I know nothing about registries so I'm not even sure what I would be looking for.

**Comment by orels1** [comment]
@JotaPe This is something we're aware of, and your observations are correct. I don't have anything to announce atm, but improving resolve performance / reducing number of refreshes is on our radar.

**Comment by jellejurre** [comment]
https://files.jellejurre.dev/ShareX/Charles_6CEw99wRSw.mp4
This also happens with adding packages and the like, here is a video showing how bad it is. Every endpoint gets hit multiple times

**Comment by orels1** [comment]
@jellejurre yes, VCC has not done any caching since essentially early in 1.x of VCC. As a result - every single time you try to install a package everything gets refreshed multiple times. This is woven pretty deeply into the core of VPM which is why it has not been addressed just yet.

However, I can share that we actually have some nice improvements to this coming in the future 2.5.0 release, which prevents constant refreshes unless the user clicks the refresh button or the cache is considered too stale. There are still extra refreshes during package additions, but I'm hoping that I can also limit those within 2.5.0 release cycle.
